src/Uno.UWP/Storage/FileIO.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using Windows.Foundation;
+using Windows.Storage.FileProperties;
+using Windows.Storage.Streams;
+
+namespace Windows.Storage
+{
+	public partial class FileIO
+
+		public static global::Windows.Foundation.IAsyncAction WriteTextAsync( global::Windows.Storage.IStorageFile file,  string contents)
+		{
+        Stream fileStream = await file.OpenStreamForWriteAsync();
+        StreamWriter streamWriter = new StreamWriter(fileStream);
+        streamWriter.Write(contents); 
```

Comment:
Could the code use asynchronous version of the API instead here? From docs - https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter.writeasync?view=netframework-4.8

Reply:
Yes, but why "await  WriteAsync" should be better than "Write"?

Reply:
It is asynchronous, so it frees up the thread to do other work instead. Async versions of IO operations are always preferable. In this case especially as if you call this method as it is currently from the UI thread, the first `await` doesn't have a `ConfigureAwait(false)` which means it will return back to the UI thread and then will perform the whole write synchronously (which could take several seconds for large files) and that will block the UI completely.